paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVEmbeddedLayoutUI.kt

```diff
@@ -0,0 +1,210 @@
+        if (displayedSavedPaymentMethod != null) {
+            SavedPaymentMethodRowButton(
+                displayableSavedPaymentMethod = displayedSavedPaymentMethod,
+                isEnabled = isEnabled,
+                isSelected = selection?.isSaved == true,
+                trailingContent = {
+                    SavedPaymentMethodTrailingContent(
+                        displayedSavedPaymentMethod = displayedSavedPaymentMethod,
+                        savedPaymentMethodAction = savedPaymentMethodAction,
+                        onViewMorePaymentMethods = onViewMorePaymentMethods,
+                        onEditPaymentMethod = onEditPaymentMethod,
+                        onManageOneSavedPaymentMethod = { onManageOneSavedPaymentMethod(displayedSavedPaymentMethod) },
+                    )
+                },
+                onClick = { onSelectSavedPaymentMethod(displayedSavedPaymentMethod) },
+                rowStyle = rowStyle
+            )
+        }
+
+        OptionalEmbeddedDivider(rowStyle)
+
+        val selectedIndex = remember(selection, paymentMethods) {
+            if (selection == null || selection.isSaved) {
+                -1
+            } else {
+                val code = selection.code()
+                paymentMethods.indexOfFirst { it.code == code }
+            }
+        }
+
+        paymentMethods.forEachIndexed { index, item ->
+            NewPaymentMethodRowButton(
+                isEnabled = isEnabled,
+                isSelected = index == selectedIndex,
+                displayablePaymentMethod = item,
+                imageLoader = imageLoader,
+                rowStyle = rowStyle
+            )
+
+            if (index != paymentMethods.lastIndex) {
+                OptionalEmbeddedDivider(rowStyle)
+            } else if (rowStyle.bottomSeparatorEnabled()) {
+                OptionalEmbeddedDivider(rowStyle)
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalEmbeddedPaymentElementApi::class)
+@Composable
+private fun OptionalEmbeddedDivider(rowStyle: Embedded.RowStyle) {
+    if (rowStyle !is Embedded.RowStyle.FloatingButton) {
+        val color = Color(rowStyle.separatorColor())
+        val thickness = rowStyle.separatorThickness()
+        val modifier = if (rowStyle is Embedded.RowStyle.FlatWithRadio) {
+            Modifier.padding(start = rowStyle.separatorInsets() + 32.dp, end = rowStyle.separatorInsets())
+        } else {
+            Modifier.padding(horizontal = rowStyle.separatorInsets())
+        }
+        Divider(
+            color = color,
+            thickness = thickness,
+            modifier = modifier
+        )
+    }
+}
+
+@OptIn(ExperimentalEmbeddedPaymentElementApi::class)
+private fun Embedded.RowStyle.bottomSeparatorEnabled(): Boolean {
+    return when (this) {
+        is Embedded.RowStyle.FloatingButton -> false
+        is Embedded.RowStyle.FlatWithRadio -> this.bottomSeparatorEnabled
```

Comment:
Is it possible to drop all the `this.` in these?

Reply:
Done

Reply:
Is this something we could lint for in future?